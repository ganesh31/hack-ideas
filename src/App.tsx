import { CodeIcon, LoginIcon } from "@heroicons/react/outline";

function App() {
  return (
    <header className="bg-slate-800 text-slate-100 flex justify-between px-4">
      {/* this div is there to bring the H1 to center */}
      <div></div>
      <h1 className="flex items-center space-x-3 text-xl font-black tracking-widest">
        <span>Hack Ideas</span>
        <CodeIcon className="w-6 h-6" />
      </h1>
      <LoginIcon data-testid="login" className=" w-8 h-8" />
    </header>
  );
}

export default App;
