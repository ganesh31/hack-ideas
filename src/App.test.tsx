import { render, screen } from "@testing-library/react";
import App from "./App";

test("renders title and login icon", () => {
  const { asFragment } = render(<App />);
  const element = screen.getByText(/Hack Ideas/i);
  const icon = screen.getByTestId("login");
  expect(icon).toBeInTheDocument();
  expect(element).toBeInTheDocument();

  expect(asFragment()).toMatchSnapshot();
});
