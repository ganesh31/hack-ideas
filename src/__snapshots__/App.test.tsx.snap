// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders title and login icon 1`] = `
<DocumentFragment>
  <header
    class="bg-slate-800 text-slate-100 flex justify-between px-4"
  >
    <div />
    <h1
      class="flex items-center space-x-3 text-xl font-black tracking-widest"
    >
      <span>
        Hack Ideas
      </span>
      <svg
        aria-hidden="true"
        class="w-6 h-6"
        fill="none"
        stroke="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10 20l4-16m4 4l4 4-4 4M6 16l-4-4 4-4"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </h1>
    <svg
      aria-hidden="true"
      class=" w-8 h-8"
      data-testid="login"
      fill="none"
      stroke="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11 16l-4-4m0 0l4-4m-4 4h14m-5 4v1a3 3 0 01-3 3H6a3 3 0 01-3-3V7a3 3 0 013-3h7a3 3 0 013 3v1"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
      />
    </svg>
  </header>
</DocumentFragment>
`;
